charts/gsp-cluster/templates/02-gsp-system/service-operator/crd.yaml

```diff
@@ -33,19 +33,147 @@ spec:
             aws:
               description: AWS specific subsection of the resource.
               properties:
-                diskSizeGB:
-                  type: integer
                 instanceType:
+                  description: InstanceType essentially defines the amount of memory
+                    and cpus on the database.
                   type: string
               type: object
+            secret:
+              description: Secret name to be used for storing relevant instance secrets
+                for further use.
+              type: string
```

Comment:
Ordinarily I'd agree with @philandstuff here. But we need to tread carefully. If a tenant defines, say, an SQSQueue named "dcs-audit" and a Postgres named "dcs-audit" the secrets need to not tread on each other.

Reply:
I agree. This is just for user benefit. Say they have a `values.yaml` file:

```yaml
secretName: cheescake
```

It will be easier for them to reference in both `Deployment` and `Postgres` kinds, to keep things in place.

```yaml
kind: Deployment
...
    env:
      - name: POSTGRES_USERNAME
        valueFrom:
          secretKeyRef:
            name: {{ .Value.secretName }}
            key: username
 ---
kind: Postgres
spec:
  secretName: {{ .Value.secretName }}
```

Also as Daniel pointed out, we don't want to get into the situation that we overwrite the secrets all he time.

Reply:
_wontfix_